pkg/attestation/crafter/runners/oidc/github_test.go

```diff
@@ -0,0 +1,348 @@
+					w.WriteHeader(http.StatusInternalServerError)
+					_, _ = w.Write([]byte("Internal Server Error"))
+				} else {
+					setupOIDCMocksHandler(w, r, privKey)
+				}
+			},
+			expectErr:         true,
+			expectErrContains: "response: 500",
+		},
+		{
+			name: "Payload decoding error",
+			mockHandler: func(w http.ResponseWriter, r *http.Request) {
+				if r.URL.Path == tokenRequestEndpoint {
+					w.Header().Set("Content-Type", "application/json")
+					_, _ = w.Write([]byte(`{"value": "not-a-token", "invalid-json`))
+				} else {
+					setupOIDCMocksHandler(w, r, privKey)
+				}
+			},
+			expectErr:         true,
+			expectErrContains: "parsing JSON",
+		},
+		{
+			name: "Token verification error - bad signature",
+			mockHandler: func(w http.ResponseWriter, r *http.Request) {
+				if r.URL.Path == tokenRequestEndpoint {
+					wrongPrivKey, _ := rsa.GenerateKey(rand.Reader, 2048)
+					wrongToken := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+					wrongToken.Header["kid"] = "test-key-id"
+					wrongSignedToken, _ := wrongToken.SignedString(wrongPrivKey)
+
+					resp := map[string]string{"value": wrongSignedToken}
+					w.Header().Set("Content-Type", "application/json")
+					_ = json.NewEncoder(w).Encode(resp)
+				} else {
+					setupOIDCMocksHandler(w, r, privKey)
+				}
+			},
+			expectErr:         true,
+			expectErrContains: "failed to verify signature",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			originalURL := os.Getenv("ACTIONS_ID_TOKEN_REQUEST_URL")
+			originalToken := os.Getenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN")
+			t.Cleanup(func() {
+				t.Setenv("ACTIONS_ID_TOKEN_REQUEST_URL", originalURL)
+				t.Setenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN", originalToken)
+			})
+			t.Setenv("ACTIONS_ID_TOKEN_REQUEST_URL", tokenRequestURL)
+			t.Setenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN", testBearerTokenValue)
+
+			if tt.setupEnv != nil {
+				tt.setupEnv(t)
+			}
+
+			if tt.mockHandler != nil {
+				mockServer.Config.Handler = tt.mockHandler
+			}
+			t.Cleanup(func() { mockServer.Config.Handler = baseHandler })
+
+			client, err := NewOIDCGitHubClient()
+			require.NoError(t, err)
+
+			// For non-error cases, we need to mock the verification function
+			if !tt.expectErr {
+				// Override the verifier function for the client
+				githubClient, ok := client.(*GitHubOIDCClient)
+				if ok {
+					// Create a custom verifier that always succeeds for our test token
+					githubClient.verifierFunc = func(ctx context.Context) (*coreoidc.IDTokenVerifier, error) {
+						provider, err := coreoidc.NewProvider(ctx, serverURL)
+						if err != nil {
+							return nil, err
+						}
+						return provider.Verifier(&coreoidc.Config{
+							SkipClientIDCheck: true,
+							SkipExpiryCheck:   true,
+							SkipIssuerCheck:   true,
+						}), nil
+					}
+				}
+			}
+
+			// Test additional methods of the client interface
+			workflowPath := client.WorkflowFilePath(context.Background())
```

Comment:
`context.Background()` is repeated multiple times, should we use a shared context or a test one?

Reply:
Done.